screwdriver/screwdriver.go

```diff
@@ -124,21 +123,22 @@ func (a api) write(url *url.URL, requestType string, bodyType string, payload io
 		return nil, fmt.Errorf("WARNING: received error from %s(%s): %v ", requestType, url.String(), err)
 	}
 
-	if res.StatusCode/100 != 2 {
-		log.Printf("WARNING: received response %d from %s ", res.StatusCode, url.String())
-		return nil, fmt.Errorf("WARNING: received response %d from %s ", res.StatusCode, url.String())
-	}
-
 	body, err := ioutil.ReadAll(res.Body)
 	if err != nil {
 		log.Printf("reading response Body from Screwdriver: %v", err)
 		return nil, fmt.Errorf("reading response Body from Screwdriver: %v", err)
 	}
 
-	parseError := json.Unmarshal(body, &errParse)
-	if parseError != nil {
-		log.Printf("unparseable error response from Screwdriver: %v", parseError)
-		return nil, fmt.Errorf("unparseable error response from Screwdriver: %v", parseError)
+	if res.StatusCode/100 != 2 {
+		var errParse SDError
+		parseError := json.Unmarshal(body, &errParse)
```

Comment:
@kkisic are we not parsing `body` into JSON any more on success ? ? Previously json.Unmarshal was happening for 2xx requests

Reply:
Because we are parsing body into SDError-format JSON, I think we should do it for non-2XX cases, not for 2XX.
[And originally this was done in the case of non-2XX.](https://github.com/screwdriver-cd/log-service/blob/v0.0.39/screwdriver/screwdriver.go#L76-L83)

If we use body in the case of 2XX, I think we need to check whether or not it is parsable in the caller of write function.